File: client_python/vehicle_data.py
#!/usr/bin/env python3
"""
Vehicle data model
Contains the current vehicle state and methods to update it
"""

class VehicleData:
    def __init__(self):
        self.speed = 0
        self.battery = 100
        self.temperature = 20
        self.direction = 'STRAIGHT'
    
    def update_from_server_data(self, parts):
        """Update data from server response"""
        if len(parts) >= 4:
            try:
                self.speed = max(0, min(100, int(parts[1])))
                self.battery = max(0, min(100, int(parts[2])))
                self.temperature = int(parts[3])
                self.direction = parts[4] if len(parts) > 4 else 'STRAIGHT'
            except (ValueError, IndexError):
                # Keep current values if there's a parsing error
                pass
    
    def get_speed_display(self):
        return f"{self.speed} km/h"
    
    def get_battery_display(self):
        return f"{self.battery}%"
    
    def get_temperature_display(self):
        return f"{self.temperature}°C"
    
    def get_direction_display(self):
        return self.direction
    
    def to_dict(self):
        """Convertir a diccionario para compatibilidad"""
        return {
            'speed': self.speed,
            'battery': self.battery,
            'temperature': self.temperature,
            'direction': self.direction
        }

File: client_java/VehicleData.java
/**
 * Modelo de datos del vehículo
 * Contiene el estado actual del vehículo y métodos para actualizarlo
 */
public class VehicleData {
    private int speed = 0;
    private int battery = 100;
    private int temperature = 20;
    private String direction = "STRAIGHT";
    
    // Getters
    public int getSpeed() { return speed; }
    public int getBattery() { return battery; }
    public int getTemperature() { return temperature; }
    public String getDirection() { return direction; }
    
    // Setters
    public void setSpeed(int speed) { 
        this.speed = Math.max(0, Math.min(100, speed)); 
    }
    
    public void setBattery(int battery) { 
        this.battery = Math.max(0, Math.min(100, battery)); 
    }
    
    public void setTemperature(int temperature) { 
        this.temperature = temperature; 
    }
    
    public void setDirection(String direction) { 
        this.direction = direction != null ? direction : "STRAIGHT"; 
    }
    
    // Actualizar desde datos del servidor
    public void updateFromServerData(String[] parts) {
        if (parts.length >= 4) {
            try {
                setSpeed(Integer.parseInt(parts[1]));
                setBattery(Integer.parseInt(parts[2]));
                setTemperature(Integer.parseInt(parts[3]));
                setDirection(parts.length > 4 ? parts[4] : "STRAIGHT");
            } catch (NumberFormatException e) {
                // Mantener valores actuales si hay error de parsing
            }
        }
    }
    
    // Formatear para display
    public String getSpeedDisplay() { return speed + " km/h"; }
    public String getBatteryDisplay() { return battery + "%"; }
    public String getTemperatureDisplay() { return temperature + "°C"; }
}
